commands/setUser.js

```diff
@@ -349,4 +349,124 @@ const updateUserNickname = async function(userid, nickname, server) {
+const toggleCommitteeOfficer = async function(assigner, assigneeInfo, server) {
+  /**
+   * Adds or removes the 'ACM Officer' and '<Committee> Officer' roles from the specified
+   * assignee based on the assigner's committee
+   *
+   * Note: this command assumes that
+   * 1) The assigner has the Committee President or PVP role
+   * 2) If the assigner is a Committee President, they have exactly one <Committee> Officer role
+   * 3) If an assignee does not have the ACM Officer role, then they also do not have a
+   *    <Committee> Officer role
+   *
+   * @param assigner - the GuildMember object representing the person calling the command
+   * @param assigneeInfo - username or userid of user whose role is to be assigned
+   * @param server - current Discord.Guild object
+   * @returns {Array} - contents: [error object, message]
+   */
+
+  // If userid not supplied, query from database
+  let assigneeId;
+  if (assigneeInfo.match('.+#([0-9]){4}')) {
+    const [username, discriminator] = assigneeInfo.split('#');
+    const db = await sqlite.open({
+      filename: config.db_path,
+      driver: sqlite3.Database,
+    });
+    let row;
+    try {
+      row = await db.get(
+        `
+        SELECT userid
+        FROM users
+        WHERE
+            username = ? AND
+            discriminator = ?
+        `,
+        [username, discriminator]
+      );
+      if (!row) return [null, 'Invalid/unverified user.'];
+      assigneeId = row.userid;
+    } catch (e) {
+      console.error(e.toString());
+      return [{ message: e.toString() }, null];
+    } finally {
+      await db.close();
+    }
+  } else {
+    if (!assigneeInfo.match('([0-9])+')) {
+      return [
+        null,
+        'Invalid user format. Please supply either a user in <username>#<discriminator> format or a user ID'
+      ];
+    }
+    assigneeId = assigneeInfo;
+  }
+
+  // Get assignee GuildMember object
+  let assignee;
+  try {
+    assignee = await server.members.fetch(assigneeId);
+  } catch(e) {
+    console.error(e);
+    return [{ message: e.toString() }, 'Invalid/unverified user.'];
+  }
+
+  // Determine assigner's committee
+  let assignerCommittee;
+  if (assigner.roles.cache.some(role => role.name === 'PVP')) {
+    // PVP assigns Board officers
+    assignerCommittee = 'Board';
+  } else {
+    for (const committee of config.committees_list) {
+      if (assigner.roles.cache.some(role => role.name === `${committee} Officer`)) {
+        assignerCommittee = committee;
+        break;
+      }
+    }
+    if (!assignerCommittee) {
+      return [
+        null,
+        'Action not permitted. If you are a committee president, make sure that you also have ' +
+        'a committee officer role before running this command.'
+      ];
+    }
+  }
+  // Officer roles to be assigned based on assigner's committee
+  const acm_officer_role = server.roles.cache.find(role => role.name === 'ACM Officer');
```

Comment:
Doesn't need to be changed now, but once my PR is also merged in, you can use the officer_role I introduced in my PR.